src/ape_ethereum/provider.py

```diff
@@ -480,8 +480,10 @@ def _eth_call(self, arguments: list) -> HexBytes:
 
         return HexBytes(result)
 
-    def _prepare_call(self, txn: TransactionAPI, **kwargs) -> list:
-        txn_dict = txn.model_dump(by_alias=True, mode="json")
+    def _prepare_call(self, txn: Union[TransactionAPI, dict], **kwargs) -> list:
+        txn_dict = (
+            txn.model_dump(by_alias=True, mode="json") if isinstance(txn, TransactionAPI) else txn
```

Comment:
mode has to do with the types it serializes too, json-compatible versus python-compatible. In this case, since are passing this data along to a request, json is required.

The main example of the difference can be seen in HexBytes types. Bytes are ok in Python but not in JSON.

Reply:
But to answer your question, yes it is a dict, but it would be a dict regardless of `mode=`'s value.
To get a str, you use `model_dump_json()`.. This is just how pydantic v2 works.

Reply:
gotcha, might be good to add a comment about why using that mode

Reply:
It is definitely the most common `mode=` used when calling this method, if you search in the whole codebase, we are doing this extensively all over. The `mode=` kwarg is also required so you have to select one or the other. If we add a comment here, we might as well comment on ever call to `model_dump` (which is a lot!) and attempt to rationalize the mode selection. I can do that, but I would like to get started on a separate PR as that is a big change.

Reply:
OK - actually it does do python by default. Still, going to need to audit for this to make sense.

Reply:
This one first: https://github.com/ApeWorX/ape/pull/2019